akka-actor/src/main/scala/akka/io/dns/DnsResourceRecords.scala

```diff
@@ -29,7 +31,7 @@ sealed abstract class ResourceRecord(val name: String, val ttl: Int, val recType
 }
 
 @ApiMayChange
-final case class ARecord(override val name: String, override val ttl: Int,
+final case class ARecord(override val name: String, override val ttl: Duration,
```

Comment:
Would need Java API as well, i.e. 
```
import akka.util.JavaDurationConverters._
def getTtl: java.time.Duration = ttl.asJava
```

Using Scala's `Duration` instead of `FiniteDuration` would complicate the Java API since there is no such thing as `Inf`. Perhaps easiest to use `FiniteDuration` with `Zero` as Never and `FiniteDuration(Long.MaxValue)` as Forever

Reply:
I think modeling `Never` and `Forever` is the problem here. The code currently uses:

```scala
  // default values (-1 and 0 are magic numbers, trust them)
  private final val Forever = -1
  private final val Never = 0
  private final val DefaultPositive = 30
```
So the arithmetic _just works_ but there's a few `if`'s here and there making the code a bit had to follow. I'm considering using:

```scala
sealed trait CacheDuration{
  val toSeconds:Long
}
case object NeverCache extends CacheDuration {
  override val toSeconds: Long = 0L // for backwards compat
}
case object CacheForever extends CacheDuration {
  override val toSeconds: Long = -1L // for backwards compat
}
case class FiniteCache(ttl:FiniteDuration) extends CacheDuration {
  require(ttl.toSeconds>0 && ttl < Duration.Inf)
  override val toSeconds: Long = ttl.toSeconds
}

object CacheDuration{
  def fromInt(n:Int):CacheDuration={ n match{
    case 0 => CacheForever
    case n if n<0 => NeverCache
    case n => FiniteCache(n.seconds)
   }
  }
}
```
And move away from `Long` or `Duration/FiniteDuration`.

Reply:
Sounds good, but I'd say:

* `CachePolicy` over `CacheDuration`
* I don't think we need the backwards compatibility (I would rather avoid it, especially with a name like `toSeconds`).  Or if we really need it maybe only in `SimpleDnsCache.Cache` itself (which is `private[io]`).